examples/Jaeger.Example.WebApi/Startup.cs

```diff
@@ -30,10 +33,16 @@ public void ConfigureServices(IServiceCollection services)
             // Adds the Jaeger Tracer.
             services.AddSingleton<ITracer>(serviceProvider =>
             {
-                string serviceName = serviceProvider.GetRequiredService<IWebHostEnvironment>().ApplicationName;
+                var serviceName = serviceProvider.GetRequiredService<IWebHostEnvironment>().ApplicationName;
+                var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
```

Comment:
just curious, if .net already has this mechanism of service providers, why did we need a custom SenderConfiguration.DefaultSenderResolver? Couldn't we register ThriftSenderFactory with service provider?

Reply:
That‘s only for ASP.NET (website stuff), but not for Desktop or Console applications. There are some dependency injection frameworks available, but I didn‘t want to force users to use one, since there is no gold standard.